infrastructure/terraform/eventviewer/main.tf

```diff
@@ -2,19 +2,38 @@ data "http" "deployjson" {
   url = "https://raw.githubusercontent.com/Azure-Samples/eventgrid-viewer-blazor/main/infrastructure/arm/azuredeploy.json"
 }
 
-resource "azurerm_template_deployment" "event_grid_deploy" {
+resource "azurerm_resource_group_template_deployment" "event_grid_deploy" {
```

Comment:
Why the name "azurerm_resource_group_template_deployment" instead of "azurerm_template_deployment" ?

Reply:
I migrated to azurerm_resource_group_template_deployment because of a warning that azurerm_template_deployment will be deprecated. See the note on top of
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/template_deployment